### PR TITLE
math: added IsPrime

### PR DESCRIPTION
**Pull request by mahmoudheshamsg**
### Description:

Added a function to the math package to check if a number is prime or not .

### Checklist:  
- [x] **Tests Passing**: Verify by running `make test`.  
- [x] **Golint Passing**: Confirm by running `make lint`.  

#### If added a new utility function or updated existing function logic:
* [x] Updated the utility package `EXAMPLES.md`
* [x] Updated the utility package `README.md`

#### If added a new utility package:
* [ ] Updated the main `README.md` utility packages table

## Summary by CodeRabbit

* **New Features**
  * Added a prime-checking capability to determine whether an integer is prime.

* **Documentation**
  * Added a usage example for the prime checker and clarified Sqrt behavior for negative inputs (now documented to return an error).

* **Tests**
  * Added tests covering edge cases and large values to validate prime-checking behavior.

### CODE REVIEW

File: math/math.go [hunk 1]
```diff
@@ -179,3 +179,34 @@ func Sqrt[T number](x T) (float64, error) {
 		z = nextZ
 	}
 }
+
+// check if a number is a prime number complexity = O(sqrt(n)
+func IsPrime(x int) (bool, error) {
+	// don't handle negative numbers
+	if x < 0 {
+		return false, errors.New("can't check on negative numbers")
```
Comment (addvanced) [inline math/math.go:187]: I feel like it's unnecessary to return errors - especially since this is the only error you return in the function.
If the number is negative, it should just return false.

File: math/EXAMPLES.md [hunk 2]
```diff
@@ -384,5 +384,33 @@ func main() {
...
+package main
+
+import (
+	"fmt"
+
+	utils "github.com/kashifkhan0771/utils/math"
+)
+
+func main() {
+	err,example := utils.IsPrime(5)
```
Comment (addvanced) [inline math/EXAMPLES.md:401]: err, example := ... doesn't match the return types of the function (bool, err).

```suggestion
	isPrime, err := utils.IsPrime(5)
```

File: math/EXAMPLES.md [hunk 3]
```diff
@@ -384,5 +384,33 @@ func main() {
...
+
+	utils "github.com/kashifkhan0771/utils/math"
+)
+
+func main() {
+	isPrime, err := utils.IsPrime(5)
+	if err != nil{
+		//handle error
+	}
+	fmt.Print(example)
```
Comment (kashifkhan0771) [inline math/EXAMPLES.md:405]: Update to `isPrime`

File: math/math.go [hunk 4]
```diff
@@ -179,3 +179,31 @@ func Sqrt[T number](x T) (float64, error) {
...
+	//since we eliminated all even numbers we can just iterate over only odd numbers
+	// I'm iterating up to sqrx+1 just to account for any rounding errors
+	for i := 3; i <= sqrx+1; i += 2 {
+		if x%i == 0 {
+			return false
+		}
+	}
+
+	return true
+}
```
[the remark this answers is not in the document]
Comment (mahmoudheshamsg) [reply math/math.go:209]: >Add negative-input test coverage for IsPrime

good suggestion will add it

**Review by kashifkhan0771 — APPROVED** [review]
Thank you @mahmoudheshamsg ❤️
